### ISSUE TITLE
Downloading files/folders shows no progress

### ISSUE THREAD

**Issue by JoshuaCrewe** [mickael-kerjean/filestash#497]
# Description of the bug

When downloading content the browser will report the download as working. It supplies details like download speed and the amount downloaded (which updated) but not how much is being downloaded total. In Firefox it is labeled `Unknown Time Left`

# Step by step instructions to reproduce the bug

1. navigate to filestash instance
2. Select a file or folder (ctl+click)
3. Click download
4. observe browser UI for download progress

# Can you replicate that error from the demo?

Yes. I have been testing with https://demo.filestash.app/files/Videos/ (the folder of Videos as well as an individual video file)

# Observed behavior

The download starts and is successful with stats on download speed and size already downloaded but not time left.

# Expected behavior

The time remaining to show in seconds/minutes left before the download is complete.

**Comment by kevinpan47** [comment]
I'm not sure what the solution could be for video files, but it is possible to view the download progress for audio files. Using `wavesurfer.js`, there is a `loading` event that lets you track the progress of the download.

In /client/pages/viewerpage/audioplayer.js, you can see this by doing something like this:
```
wavesurfer.current.on("loading", (percentLoaded) => {
    console.log(percentLoaded);
})
```
This event fires continuously when loading, with the percentage of the file downloaded (between 0-100).

**Comment by kevinpan47** [comment]
@mickael-kerjean I saw your comments in the PR for the spinner display, I actually had the same idea and mentioned it in this issue here.

**Comment by mickael-kerjean** [comment]
yep I have a proto running in my machine which uses that, will probably be in Filestash either today or tomorrow

**Comment by kevinpan47** [comment]
@mickael-kerjean
A similar event exists for videos loaded with videojs, you can try it out by putting this in the `useEffect` hook in client/pages/viewerpage/videoplayer.js:
```
player.on("progress", () => {
    const loadedPercent = Math.trunc(player.bufferedPercent() * 100);
    console.log(loadedPercent);
});
```
The UI can then display the loading progress, however, videos are often only partially loaded, so the initial loading won't go from 0 to all the way to 100, it'll stop somewhere before that:
![image](https://user-images.githubusercontent.com/71465502/231580045-6c0b8d80-20fa-4854-b75b-2bdd258e647a.png)

**Comment by mickael-kerjean** [comment]
yes but things aren't as simple:
- there's a few other improvements in the audio player that I always wanted to work on, things like being able to stream things over using a chromecast and I'm already focusing on that now, improvements in the video is in the pipeline but not until all that work is done
- videojs will probably gonna go away if the poc using the base video tag works
